payment/payment.go

```diff
@@ -38,10 +38,8 @@ func FromPublicKey(pubkey *btcec.PublicKey, network *network.Network) *Payment {
 
 // FromPayment creates a Payment struct from a another Payment
 func FromPayment(payment *Payment) *Payment {
-	buf := payment.Script
-	hash := hash160(buf)
-	redeem := &Payment{payment.Network, payment.PublicKey, hash, nil, nil, nil}
-	return &Payment{nil, nil, nil, nil, redeem, nil}
+	redeem := &Payment{payment.Network, payment.PublicKey, payment.Hash, nil, nil, nil}
```

Comment:
`redeem` looks better now, but I would copy all the fields (even if `nil`) rahter than forcing them to `nil`:
```suggestion
	redeem := &Payment{payment.Network, payment.PublicKey, payment.Hash, payment.BlindingKey, payment.Redeem, payment.Script}
```